client/src/components/timeline/diff/header-revision.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import RevisionEntity from 'app/entities/revision';
+import Header from 'app/components/timeline/header';
+import Spinner from 'app/components/timeline/spinner';
+
+class HeaderRevision extends React.PureComponent {
+
+	componentDidMount() {
+		if ( !this.props.revision ) {
+			return;
+		}
+
+		if ( this.props.revision.meta.status === 'ready' ) {
+			this.props.fetchRevision( this.props.revision.id );
+		}
+	}
+
+	render() {
+		if ( !this.props.revision ) {
+			return (
+				<Header />
+			);
+		}
+
+		if ( this.props.revision.meta.status === 'ready' ) {
+			return (
+				<Header href={this.props.url} />
+			);
+		}
+
+		if ( this.props.revision.meta.status === 'fetching' ) {
```

Comment:
Is the revision status check necessary considering that the Header is part of the Diff Container? Can the diff be ready without the revision being ready? Same goes for error

Reply:
@dayllanmaza yes, because there are two revisions, and you may not have already downloaded the other revision. Really the only reason I need the other revision is for the timestamp (which is not returned with the diff). But yeah it might not even be one of the two users.